lib/rig_inbound_gateway_web/connection_init.ex

```diff
@@ -47,10 +49,35 @@ defmodule RigInboundGatewayWeb.ConnectionInit do
            Subscriptions.from_json(request.body),
          subscriptions = Enum.uniq(jwt_subs ++ query_subs),
          :ok <- SubscriptionAuthZ.check_authorization(request) do
+
+      {:ok, vconnection_pid} = Codec.deserialize(request.connection_token)
+      |> case do
+        {:error, nil} ->
+          # We're going to accept the connection
+          # and let VConnection handle subscription refresh and heartbeat
+          VConnection.start(self(), subscriptions, heartbeat_interval_ms, subscription_refresh_interval_ms)
+        {:ok, pid} ->
+          # We're going to try to reconnect
+          if Process.alive?(pid) do
+            # Try a reconnect
+            vpid = GenServer.call(pid, :reconnect)
+
+            # Re-register subscriptions
+            send pid, :set_subscriptions
```

Comment:
if the client is reconnecting and passes a different set of subscriptions, the newer set should replace the existing subscriptions